mypy/semanal_pass3.py

```diff
@@ -431,12 +438,28 @@ def check_for_omitted_generics(self, typ: Type) -> None:
             if t.type_of_any == TypeOfAny.from_omitted_generics:
                 self.fail(messages.BARE_GENERIC, t)
 
-    def fail(self, msg: str, ctx: Context, *, blocker: bool = False) -> None:
+    def lookup_qualified(self, name: str, ctx: Context,
+                         suppress_errors: bool = False) -> Optional[SymbolTableNode]:
+        return self.sem.lookup_qualified(name, ctx, suppress_errors=suppress_errors)
+
+    def lookup_fully_qualified(self, fullname: str) -> SymbolTableNode:
+        return self.sem.lookup_fully_qualified(fullname)
+
+    def dereference_module_cross_ref(
+            self, node: Optional[SymbolTableNode]) -> Optional[SymbolTableNode]:
+        return self.sem.dereference_module_cross_ref(node)
+
+    def fail(self, msg: str, ctx: Context, serious: bool = False, *,
+             blocker: bool = False) -> None:
+        # TODO: Process all arguments
         self.errors.report(ctx.get_line(), ctx.get_column(), msg)
 
     def fail_blocker(self, msg: str, ctx: Context) -> None:
         self.fail(msg, ctx, blocker=True)
 
+    def note(self, msg: str, ctx: Context) -> None:
+        self.sem.note(msg, ctx)
```

Comment:
What is the reason we can reuse `self.sem.note` here but not `self.sem.fail` above?

Reply:
These were different originally, but it seems like there's no good reason. I'll unify them.